equinox/nn/attention.py

```diff
@@ -0,0 +1,136 @@
+from typing import Optional
+
+import jax
+import jax.numpy as jnp
+import jax.random as jrandom
+import numpy as np
+
+from ..custom_types import Array
+from ..module import Module, static_field
+from .dropout import Dropout
+from .linear import Linear
+
+
+class MultiheadAttention(Module):
+    """
+    Multihead Attention layer from 'Attention Is All You Need' (https://arxiv.org/abs/1706.03762)
+    """
```

Comment:
Would it be possible to add the mathematical formulation here?

PyTorch kind-of do this for attention [here](https://pytorch.org/docs/stable/generated/torch.nn.MultiheadAttention.html#torch.nn.MultiheadAttention), but I'm imagining something more precise, a la [here](https://pytorch.org/docs/stable/generated/torch.nn.Conv1d.html#torch.nn.Conv1d).

The documentation generation supports LaTeX: `stuff $\alpha$ morestuff`.

Reply:
Added documentation in the latest commit - though I had to skip using flake8 because it's complaining about the '\i' and '\s' in '\intercal' and '\sqrt'. How can this be ignored by the checks in this repo?

Reply:
This is because Python/flake8 is trying to interpret those as escape codes, like `\n` for a newline. Parsing of escape codes can be disabled by prefixing an `r` before the string, i.e.: `r""" ... stuff ... """`